packages/core/data-transfer/src/engine/index.ts

```diff
@@ -647,11 +642,12 @@ class TransferEngine<
       );
     }
 
-    const { sourceSchema, destinationSchema } = await this.#getSchemas();
+    const sourceSchemas = (await this.sourceProvider.getSchemas?.()) as SchemaMap;
+    const destinationSchemas = (await this.destinationProvider.getSchemas?.()) as SchemaMap;
```

Comment:
```suggestion
    const sourceSchemas = (await this.sourceProvider.getSchemas?.()) as SchemaMap | undefined;
    const destinationSchemas = (await this.destinationProvider.getSchemas?.()) as SchemaMap | undefined;
```

Reply:
this shouldn't be like this, we should use this.#getSchemas() it's saving the schemas in memory